src/io/utils.js

```diff
@@ -17,16 +18,26 @@ export function walkPromise(curPath) {
     return lstatPromise(curPath)
       .then((stat) => {
         if (stat.isFile()) {
-          var { size } = stat;
-          result[path.relative(basePath, curPath)] = { size };
+          const relPath = path.relative(basePath, curPath);
+          const { size } = stat;
+          if (shouldIncludePath(relPath)) {
+            result[relPath] = { size };
+          } else {
+            log.debug(`Skipping file: ${relPath}`);
+          }
         } else if (stat.isDirectory()) {
           return readdirPromise(curPath)
             .then((files) => {
               // Map the list of files and make a list of readdir
               // promises to pass to Promise.all so we can recursively
               // get the data on all the files in the directory.
               return Promise.all(files.map((fileName) => {
-                return walk(path.join(curPath, fileName));
+                if (shouldIncludePath(fileName)) {
```

Comment:
Mann, the recursion was hurting my brain on this. You're right, I just pushed a change that only calls `shouldIncludePath()` once, right at the top.